### DISCUSSION TITLE
set individual values of the RtcDateTime object

### DISCUSSION THREAD

**Discussion by Iceteavanill** [Makuna/Rtc#181] [Q&A] [answered]
Is it possible to set individual values in the RtcDateTime object. As far as I understand this library it is possible to increment or decrement the amount of seconds but is it possible to increment or decrement the month or days value directly? Adding a fixed value of seconds to the object wouldn't increment the months reliably due to the different lengths during the year. And the _month value of the Object is protected
My end goal is to set time via push-buttons and incrementing the time / dates individually. Am I missing something or is there a workaround?

**Comment by Makuna** [comment, marked as answer]
Nope, this is on purpose.  Its an optimization.  Just construct a temp.  When the compiler knows its members are not directly accessible it can do some optimizations.  Further, there are details for incrementing that can cause invalid dates if you use daylight shifting (which I do not recommend except where you display or take input).  But adding seconds will never do this.

Take this example which does compile successfully and works for most cases, as it assumes you want the last day of the next month if the day is beyond a valid day of the month.
NOTE:  DaysInMonth is coming shortly to wrap the access to progmem c_daysInMonth for easier readability
```
                    uint16_t year = when.Year();
                    uint8_t month = when.Month() + 1;
                    uint8_t day = when.Day();

                    if (month > 12)
                    {
                        year++;
                        month = 1;
                    }
                    uint8_t daysInMonth = RtcDateTime::DaysInMonth(year, month);
                    if (day > daysInMonth)
                    {
                        day = daysInMonth;
                    }
                    RtcDateTime next(year,
                        month,
                        day,
                        when.Hour(),
                        when.Minute(),
                        when.Second());
```

**Reply by Makuna** [reply]
If you need this do an alarm feature, you may want to hold off, I am actively working on an RtcAlarmManager at the moment which handles a monthly alarm that honors a 31/30/29/28 day of the month as last day of the month if less days in the month happen for a specific month but doesn't force all months to be 30/29/28 when you select 31st.
I am writing tests for it now.

**Comment by Iceteavanill** [comment]
I am not looking for a alarm function but thanks for the heads up. So the easiest method is use a placeholder for year, months. For days, minutes and seconds I can just add seconds via the mathematical operator and the object handles the rest for me right?

**Reply by Makuna** [reply]
correct.
